src/bfs/rmat_edge_generator/generate_edge_list.cpp

```diff
@@ -98,21 +141,39 @@ int main(int argc, char **argv) {
   std::string out_edge_list_file_name;
   parse_options(argc, argv, &rmat_option, &out_edge_list_file_name);
 
-  rmat_edge_generator rmat(rmat_option.seed, rmat_option.vertex_scale, rmat_option.edge_count,
-                           rmat_option.a, rmat_option.b, rmat_option.c,
-                           1.0 - (rmat_option.a + rmat_option.b + rmat_option.c),
-                           rmat_option.scramble_id, rmat_option.generate_both_directions);
+  size_t num_generated_edges = 0;
 
-  std::ofstream edge_list_file(out_edge_list_file_name);
-  if (!edge_list_file.is_open()) {
-    std::cerr << "Cannot open " << out_edge_list_file_name << std::endl;
-    std::abort();
+#ifdef _OPENMP
+#pragma omp parallel reduction(+:num_generated_edges)
+#endif
+  {
+    const int num_edges_per_thread = num_local_edges(0, rmat_option.edge_count - 1, thread_num(), num_threads());
+    const uint64_t seed = rmat_option.seed + thread_num();
+
+    rmat_edge_generator rmat(seed, rmat_option.vertex_scale, num_edges_per_thread,
+                             rmat_option.a, rmat_option.b, rmat_option.c,
+                             1.0 - (rmat_option.a + rmat_option.b + rmat_option.c),
+                             rmat_option.scramble_id, rmat_option.generate_both_directions);
+
+    std::ofstream edge_list_file(out_edge_list_file_name + "_" + std::to_string(thread_num()));
+    if (!edge_list_file.is_open()) {
+      std::cerr << "Cannot open " << out_edge_list_file_name << std::endl;
+      std::abort();
+    }
+
+    for (auto edge : rmat) {
+      edge_list_file << edge.first << " " << edge.second << "\n";
+      ++num_generated_edges;
+    }
+    edge_list_file.close();
   }
 
-  for (auto edge : rmat) {
-    edge_list_file << edge.first << " " << edge.second << "\n";
+  // Sanitary check
+  const size_t num_edges_supposed_to_generate = (rmat_option.generate_both_directions) ? rmat_option.edge_count * 2ULL : rmat_option.edge_count;
+  if (num_edges_supposed_to_generate != num_generated_edges) {
+    std::cerr << "The number of generated edges is wrong " << num_generated_edges << std::endl;
```

Comment:
add in the error message the number of edges that were supposed to have been generated so the numbers can be compared for debugging.